File: GOF/Adapter/cpp/Coord.cpp
//
// Created by 17271 on 2019/3/27.
//

#include <xutility>
#include "Coord.h"

Coord::Coord(int x, int y)
	: m_x{x}, m_y{y}
{}

Coord operator+(Coord &a, Coord &b)
{
	return std::move(Coord(a.m_x + b.m_x, a.m_y + b.m_y));
}

File: GOF/Adapter/java/src/Coord.java
public class Coord {

	private int mX;
	private int mY;

	public Coord() {}

	public Coord(int x, int y) {
		mX = x;
		mY = y;
	}

	public Coord add(Coord a) {
		return new Coord(mX + a.mX, mY + a.mY);
	}
}
